shared/lsif/ranges.ts

```diff
@@ -321,6 +348,19 @@ export async function rangesInRangeWindow(
     )
 }
 
+/** Retrieve all ranges in this document that contain code intelligence. */
+export async function underlinesInDocument(
```

Comment:
We explicitly cache range responses in memory because it's an expensive operation. Could we share the same code path as we use in the defs/hovers paths?